src/Medulla/Kubernetes/Operator/Entities/V1Alpha1DataEntity.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Licensed to the Medulla Contributors under one or more agreements.
+// The Medulla Contributors licenses this file to you under the Apache 2.0 license.
+// See the LICENSE file in the project root for more information.
+
+using k8s.Models;
+using KubeOps.Operator.Entities;
+using KubeOps.Operator.Entities.Annotations;
+
+namespace DatabaseControllerKubeOps.Controller.Entities;
+
+
+/// <summary>
+/// DatabaseSpec is spec of CRD for database
+/// </summary>
+public struct DatabaseSpec
+{
+    [Required]
+    public string? Name { get; set; }
+    [Required]
+    public string? Host { get; set; }
+    [Required]
+    public string? Dialect { get; set; }
+    [Required]
+    public string? UsernameSecretName { get; set; }
+    [Required]
+    public string? PasswordSecretName { get; set; }
+    [Required]
+    public List<ModelSpec> Models { get; set; }
+}
+
+/// <summary>
+/// Spec of CRD for model within Database
+/// </summary>
+public struct ModelSpec
+{
+    [Required]
+    public string? Name { get; set; }
+    [Required]
+    public List<ColumnSpec>? Columns { get; set; }
+}
+
+
+
+/// <summary>
+/// Spec in Column for validation
+/// </summary>
+public struct ValidateSpec
+{
+    public string? Is { get; set; }
+    public string? Not { get; set; }
+    public bool? IsEmail { get; set; }
+    public bool? IsUrl { get; set; }
+    public bool? IsIp { get; set; }
+    public bool? IsIpV4 { get; set; }
+    public bool? IsIpV6 { get; set; }
+    public bool? IsAlpha { get; set; }
+    public bool? IsAlphaNumeric { get; set; }
+    public bool? IsNumeric { get; set; }
+    public bool? IsInt { get; set; }
+    public bool? IsFloat { get; set; }
+    public bool? IsDecimal { get; set; }
+    public bool? IsLowercase { get; set; }
+    public bool? IsUppercase { get; set; }
+    public bool? IsNull { get; set; }
+    public bool? NotNull { get; set; }
+    public bool? NotEmpty { get; set; }
+}
+
+/// <summary>
+/// Spec of Column within Model as array
+/// </summary>
+public struct ColumnSpec
+{
+    public string? ColumnName { get; set; }
+    public string? Type { get; set; }
+    public bool AllowNull { get; set; }
+    public string? DefaultValue { get; set; }
+    public bool AutoIncrement { get; set; }
+    public bool PrimaryKey { get; set; }
+    public string? Field { get; set; }
+    public string? Unique { get; set; }
+    public string? Comment { get; set; }
+    public ValidateSpec? Validate { get; set; }
+
+}
```

Comment:
Avoid sticking multiple structs / classes in a single file.

If you have a struct named `ColumnSpec` it should be in a file named `ColumnSpec.cs`. Its very common to reuse structs across multiple CRDs so we shouldn't group them under a single file.

We are also missing XML comments for all the public members here.

Reply:
Moved struct usage into a new model/database directory:  https://github.com/recro/medulla/pull/153/commits/a402498366570dc2946880699739e0fabef958b5